timemachine/fe/dummy.py

```diff
@@ -441,45 +457,58 @@ def flag_bonds(mol, core, bond_idxs):
     dummy_ixns = make_bond_set(dummy_ixns)
 
     # 2. process dummy bonds
-    dgs, ags, ag_ixns = generate_optimal_dg_ag_pairs(mol, core, bond_idxs)
+    dgs, ags, ag_ixns = generate_optimal_dg_ag_pairs(core, valence_idxs)
 
     allowed_ixns = set()
 
     for ixns in ag_ixns:
         allowed_ixns |= ixns
 
-    for b_idx, atom_idxs in enumerate(bond_idxs):
+    for b_idx, atom_idxs in enumerate(valence_idxs):
         if tuple(ordered_tuple(atom_idxs)) in allowed_ixns:
             keep_flags[b_idx] = 1
 
     return keep_flags
 
 
-def generate_dg_ag_pairs(mol, core, bond_idxs):
+class MissingBondError(Exception):
+    pass
```

Comment:
The usage of this makes it seem like a more appropriate name might be something like `MissingAtomError` or `InvalidBondError`? If it doesn't make sense to include extra attributes (e.g. the invalid `i` or `j` value), would it make sense to just use `ValueError` with a message? (Could also see adding `i` and `j` fields, which could justify a custom error type)

Reply:
Hm the name is correct, its indicating that a bond is missing (i.e. we have a disconnected atom). It's not clear *which* bond is missing unfortunately (since an arbitrary number of bonds may be possibe here).

Reply:
Also: `ValueError` is harder to test (since we'd need to match the resulting error string some how)

Reply:
> Hm the name is correct, its indicating that a bond is missing

Ah, sorry, I misunderstood. Looking again at `generate_dg_ag_ixns`, this makes sense.

Would it make sense to include a message and/or the missing indices as attributes on `MissingBondError`?

Reply:
I think that's totally reasonable!

Reply:
This has been modified to:

``` python
class MissingBondError(Exception):
    def __init__(self, bad_atom):
        message = "atom " + str(bad_atom) + " is disconnected."
        super().__init__(message)
```